### DISCUSSION TITLE
Nested Dynamic Routes Layout

### DISCUSSION THREAD

**Discussion by maxtotheguenther** [vercel/next.js#51038] [App Router]
Hey Guys,

I want to create the following urls for my website

**Category Urls**
Within those Urls i want to share a persistent category header.
/gaming
/gaming/gear
/gaming/professionals

**Subcategory Urls**
Within those Urls i want to share a persitent subcategory header. So i want to get rid ot the category header and show a dedicated subcategory header
/gaming/fortnite
/gaming/fortnite/gear
/gaming/fortnite/professionals

I have the following routing folder structure. 

📁[categorySlug]
  - ⏺️ layout.tsx
  - ⏺️ page.tsx
  - 📁gear
      - ⏺️page.tsx
  - 📁professionals
      - ⏺️page.tsx
  - 📁[subcategorySlug]
      - ⏺️ layout.tsx
      - ⏺️ page.tsx
      - 📁gear
           - ⏺️page.tsx
       - 📁professionals
           - ⏺️page.tsx

Is there a more effective method to achieve the desired outcome, considering that the category header will inevitably appear above the subcategory header?

**Comment by SamKomesarook** [comment]
I think what you've got is the best way to go about what you're trying to do. The first catchall will cover the category header, and the same with the subheader. If the category and subcategory slugs are defined, you could consider using the [generateStaticParams](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#generating-static-params) to build them.
